machine/v1beta1/types_azureprovider.go

```diff
@@ -0,0 +1,191 @@
+
+	// SpotVMOptions allows the ability to specify the Machine should use a Spot VM
+	SpotVMOptions *SpotVMOptions `json:"spotVMOptions,omitempty" protobuf:"bytes,22,opt,name=spotVMOptions"`
+
+	// SecurityProfile specifies the Security profile settings for a virtual machine.
+	// +optional
+	SecurityProfile *SecurityProfile `json:"securityProfile,omitempty" protobuf:"bytes,23,opt,name=securityProfile"`
+}
+
+// SpotVMOptions defines the options relevant to running the Machine on Spot VMs
+type SpotVMOptions struct {
+	// MaxPrice defines the maximum price the user is willing to pay for Spot VM instances
+	MaxPrice *resource.Quantity `json:"maxPrice,omitempty" protobuf:"bytes,1,opt,name=maxPrice"`
+}
+
+// AzureMachineProviderStatus is the type that will be embedded in a Machine.Status.ProviderStatus field.
+// It contains Azure-specific status information.
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type AzureMachineProviderStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// VMID is the ID of the virtual machine created in Azure.
+	// +optional
+	VMID *string `json:"vmId,omitempty" protobuf:"bytes,2,opt,name=vmId"`
+
+	// VMState is the provisioning state of the Azure virtual machine.
+	// +optional
+	VMState *VMState `json:"vmState,omitempty" protobuf:"bytes,3,opt,name=vmState,casttype=VMState"`
+
+	// Conditions is a set of conditions associated with the Machine to indicate
+	// errors or other status.
+	// +optional
+	Conditions []AzureMachineProviderCondition `json:"conditions,omitempty" protobuf:"bytes,4,rep,name=conditions"`
+}
+
+// VMState describes the state of an Azure virtual machine.
+type VMState string
+
+var (
+	// ProvisioningState related values
+	// VMStateCreating ...
+	VMStateCreating = VMState("Creating")
+	// VMStateDeleting ...
+	VMStateDeleting = VMState("Deleting")
+	// VMStateFailed ...
+	VMStateFailed = VMState("Failed")
+	// VMStateMigrating ...
+	VMStateMigrating = VMState("Migrating")
+	// VMStateSucceeded ...
+	VMStateSucceeded = VMState("Succeeded")
+	// VMStateUpdating ...
+	VMStateUpdating = VMState("Updating")
+
+	// PowerState related values
+	// VMStateStarting ...
+	VMStateStarting = VMState("Starting")
+	// VMStateRunning ...
+	VMStateRunning = VMState("Running")
+	// VMStateStopping ...
+	VMStateStopping = VMState("Stopping")
+	// VMStateStopped ...
+	VMStateStopped = VMState("Stopped")
+	// VMStateDeallocating ...
+	VMStateDeallocating = VMState("Deallocating")
+	// VMStateDeallocated ...
+	VMStateDeallocated = VMState("Deallocated")
+
+	// VMStateUnknown ...
+	VMStateUnknown = VMState("Unknown")
+)
+
+// Image is a mirror of azure sdk compute.ImageReference
+type Image struct {
+	// Fields below refer to os images in marketplace
```

Comment:
Explain each field, when user invokes `oc explain` the description has to be per-field.